src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -305,8 +305,7 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
       !$omp do schedule(runtime)  &
       !$omp private(invAreaCell, k33Norm, k, dzTop, dTdzTop, dSdzTop, i, iEdge, cell1, cell2, &
       !$omp         iCellSelf, dcEdgeInv, areaEdge, drhoDT, drhoDS, dTdx, dSdx, drhoDx, &
-      !$omp         slopeTaperUp, slopeTaperDown, sfcTaperUp, sfcTaperDown, sfcTaper) &
-      !$omp firstprivate(slopeTriadUp, slopeTriadDown)
```

Comment:
Array variables are not supposed to be in a private clause. I confirmed this change with @mattdturner. I'm surprised this had passed a threading bfb test.

Reply:
be a little careful with the language here - arrays can certainly be in private clauses if they are truly private to each loop iteration (eg the dzTop, etc arrays).  And in the old threading model, all local arrays ended up being thread private which was part of the problem.  Having not looked carefully at slopeTriadUp/Down, it's not completely obvious - these look like they are large-ish local copies replicated across icell iterations. (k,2,nEdges) for each iCell iteration? Could be wrong

Reply:
These are registry arrays that have dimensions (k, 2, nEdges) - it wouldn't make sense for these to be thread private. They are used later, and are not just a local copy.